Sources/TSCBasic/Path.swift

```diff
@@ -1,26 +1,178 @@
 /*
  This source file is part of the Swift.org open source project
 
- Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+ Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
  Licensed under Apache License v2.0 with Runtime Library Exception
 
  See http://swift.org/LICENSE.txt for license information
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 */
-#if os(Windows)
-import Foundation
-import WinSDK
-#endif
-
-#if os(Windows)
-private typealias PathImpl = UNIXPath
-#else
-private typealias PathImpl = UNIXPath
-#endif
+import SystemPackage
 
 import protocol Foundation.CustomNSError
 import var Foundation.NSLocalizedDescriptionKey
 
+public protocol Path: Hashable, Codable, CustomStringConvertible {
```

Comment:
wonder if we should pick a different name for these types, as ideally we deprecate them all over time

Reply:
Use another name in #219